pkg/bee/client.go

```diff
@@ -433,25 +432,13 @@ func (c *Client) UploadBytes(ctx context.Context, b []byte, o api.UploadOptions)
 }
 
 // UploadChunk uploads chunk to the node
-func (c *Client) UploadChunk(ctx context.Context, chunk *Chunk, o api.UploadOptions) (err error) {
-	p := bmtlegacy.NewTreePool(chunkHahser, swarm.Branches, bmtlegacy.PoolSize)
-	hasher := bmtlegacy.New(p)
-	err = hasher.SetSpan(int64(chunk.Span()))
+func (c *Client) UploadChunk(ctx context.Context, chunk *Chunk, o api.UploadOptions) (swarm.Address, error) {
```

Comment:
As the chunk address is not needed to Chunks.Upload, this method UploadChunk can just receive chunk data as []byte instead the *Chunk.